test/acceptance/cli-test.js

```diff
@@ -1317,6 +1323,52 @@ describe('ember-template-lint executable', function () {
         expect(result.stderr).toBeFalsy();
       });
     });
+
+    describe('with --max-warnings param', function () {
+      it('should should return exit code 1 if total error is less than max-warnings', async function () {
+        project.setConfig({
+          rules: {
+            'no-bare-strings': true,
+            'no-html-comments': true,
+          },
+        });
+        project.write({
+          app: {
+            templates: {
+              'application.hbs':
+                '<h2>Here too!!</h2><div>Bare strings are bad...</div><!-- bad html comment! -->',
+            },
+          },
+        });
+
+        let result = await run(['.', '--max-warnings=3']); // One less than total
+
+        expect(result.exitCode).toEqual(1);
+        expect(result.stderr).toBeFalsy();
+      });
+
+      it('should return exit code 0 if warnings is less or equal to max-warnings', async function () {
+        project.setConfig({
+          rules: {
+            'no-bare-strings': true,
+            'no-html-comments': true,
+          },
+        });
+        project.write({
+          app: {
+            templates: {
+              'application.hbs':
+                '<h2>Here too!!</h2><div>Bare strings are bad...</div><!-- bad html comment! -->',
+            },
+          },
+        });
+
+        let result = await run(['.', '--max-warnings=4']);
+
+        expect(result.exitCode).toEqual(0);
+        expect(result.stderr).toBeFalsy();
+      });
```

Comment:
This test doesn't seem to be testing the thing it says it is.

I think you need to tweak it like:

```suggestion
      it('should return exit code 0 if warnings is less or equal to max-warnings', async function () {
        project.setConfig({
          rules: {
            'no-bare-strings': 'warn',
            'no-html-comments': 'warn',
          },
        });
        project.write({
          app: {
            templates: {
              'application.hbs':
                '<h2>Here too!!</h2><div>Bare strings are bad...</div><!-- bad html comment! -->',
            },
          },
        });

        let result = await run(['.', '--max-warnings=3']);

        expect(result.exitCode).toEqual(0);
        expect(result.stderr).toBeFalsy();
      });

      it('should return exit code 1 if warnings is greater than max-warnings', async function () {
        project.setConfig({
          rules: {
            'no-bare-strings': 'warn',
            'no-html-comments': 'warn',
          },
        });
        project.write({
          app: {
            templates: {
              'application.hbs':
                '<h2>Here too!!</h2><div>Bare strings are bad...</div><!-- bad html comment! -->',
            },
          },
        });

        let result = await run(['.', '--max-warnings=2']);

        expect(result.exitCode).toEqual(1);
        expect(result.stderr).toMatchInlineSnapshot();
      });
```